src/login.tsx

```diff
@@ -30,70 +36,135 @@ const Login = () => {
+      setLoading(true);
+      callLogin(email, password);
+    }
   };
+  
+  const INCREMENT_COUNTER = gql`
+    mutation ($email: String!, $password: String!){
+      login(data: {email: $email, password: $password}){
+        token
+        user{
+          id
+          name
+        }
+      }
+    }
+  `;
+
+  const [mutateFunction,  { loading, error }] = useMutation(INCREMENT_COUNTER);
+
+  const callLogin = (email, password) => {
+    return (
+      mutateFunction({
+        variables: {email: email, password: password}
+      })
+      .then(resp => {
+        if(resp.data == null){
+          setLoginError('Email ou senha inválidos');
+          saveUserToken('');
+          setLoading(false);
+        } else {
+          saveUserToken(resp.data.login.token);
+          setLoading(false);
+          Navigation.push(props.componentId, {
+            component: {
+              name: 'Home',
+              options: {
+                topBar: {
+                  title: {
+                    text: 'Home'
+                  }
+                }
+              }
+            }
+          });
+        }
+      })
+      .catch((e) => {
+        setLoginError(e.message);
+        setLoading(false);
+      })
+    );
+  }
 
   return (
-    <Background>
-      <MainText>Bem-vind@ à Taqtile!</MainText>
-      <FieldsCell>
-        <View>
-          <SubText>Email</SubText>
-          <TextBox
-            placeholder={'exemple@exemple.com'}
-            onChangeText={(value) => setEmail(value)}
-            defaultvalue={email}
-            status={emailError}
-          />
-          {emailError === 'empty' ? MandatoryFieldWarning :
-            emailError === 'structure' ? InvalidEmailWarning : NoErrorText
-          }
-        </View>
-        <View>
-          <SubText>Senha</SubText>
-          <TextBox
-            secureTextEntry={true}
-            onChangeText={(value) => setPassword(value)}
-            defaultvalue={password}
-            status={passwordError}
-          />
-          {passwordError === 'empty' ? MandatoryFieldWarning :
-            passwordError === 'structure' ? InvalidPasswordWarning :
-            passwordError === 'length' ? PasswordlengthWarning: NoErrorText
-          }
-        </View>
-        <LoginButton>
-          <ButtonText onPress={() => validateLogin()}>Entrar</ButtonText>
-        </LoginButton>
-      </FieldsCell>
-    </Background>
+    <View>
+      <Background>
+        <MainText>Bem-vind@ à Taqtile!</MainText>
+        <FieldsCell>
+          <View>
+            <SubText>Email</SubText>
+            <TextBox
+              placeholder={'email@exemple.com'}
+              autoCapitalize='none'
+              onChangeText={(value) => setEmail(value)}
+              defaultvalue={email}
+              status={emailError}
+            />
+            {emailError === 'empty' ? MandatoryFieldWarning :
+              emailError === 'structure' ? InvalidEmailWarning : NoErrorText
+            }
```

Comment:
esse `NoErrorText` é necessário? pelo que eu vi ele n tem nada dentro ai eu acho que vc poderia usar conditional rendering aqui

```suggestion
            {emailError === 'empty' && MandatoryFieldWarning}
            {emailError === 'structure' && InvalidEmailWarning}
```

Reply:
Então, eu coloquei o texto vazio para o react reconhecer um componente naquele lugar. Se eu passasse apenas <></> ele removia o espaço da mensagem quando não tem mensagem para ser mostrada e mudaria a posição dos outros componentes. Mas darei uma olhada nesse conditional rendering!

Reply:
Opa, testei aqui e realmente, tem o mesmo problema do espaçamento quando o componente some. Dessa forma fica bem mais legível que vários if ternários em serie, então só adicionei `{ !emailError && NoErrorText }` para o caso do texto vazio. Não sei de nenhuma outra forma de contornar esse problema.